api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -381,47 +415,73 @@ private void validateAreaOfResearch(Profile profile) throws BadRequestException
    * object.
    *
    * @param updatedProfile
-   * @param prevProfile
+   * @param previousProfile
    * @throws BadRequestException
    */
   @VisibleForTesting
-  public void validateProfile(@Nonnull Profile updatedProfile, @Nullable Profile prevProfile)
+  public void validateProfile(
+      @Nonnull Profile updatedProfile, @Nullable Profile previousProfile, boolean userIsAdmin) {
+
+    final boolean isNewProfile = previousProfile == null;
+    final Diff diff = javers.compare(previousProfile, updatedProfile);
+
+    validateProfileForCorrectness(updatedProfile, isNewProfile, diff);
+
+    if (userIsAdmin) {
+      validateChangesAllowedByAdmin(diff);
+    } else {
+      validateChangesAllowedByUser(diff);
+    }
+  }
+
+  private void validateProfileForCorrectness(Profile profile, boolean isNewProfile, Diff diff)
       throws BadRequestException {
-    boolean isNewObject = prevProfile == null;
-    Diff diff = javers.compare(prevProfile, updatedProfile);
+    if (isNewProfile || fieldChanged(diff, "username")) {
+      validateUsername(profile);
+    }
+    if (isNewProfile || fieldChanged(diff, "contactEmail")) {
+      validateContactEmail(profile);
 
-    if (!getChangesWithPrefix(diff, "username").isEmpty() || isNewObject) {
-      validateUsername(updatedProfile);
+      // only validate if the new profile has an affiliation - some older users do not
+      if (profile.getVerifiedInstitutionalAffiliation() != null) {
+        validateInstitutionalAffiliationAgainstEmail(profile);
+      }
     }
-    if (!getChangesWithPrefix(diff, "contactEmail").isEmpty() || isNewObject) {
-      validateContactEmail(updatedProfile);
+    if (isNewProfile || fieldChanged(diff, "givenName")) {
```

Comment:
This would be cleaner if you passed in `isNewProfile` to `fieldChanged()`. Actually I'd probably have a method that absorbed all the `if`s as well so there's not so much branching. Like

`boolean validateIfChanged(Diff diff, String fieldName, Consumer<Profile> validateFunction);` 

Bonus points for putting all that into an enum, or using a map from field names to validators.